### ISSUE TITLE
Can not read XMP data from PNG

### ISSUE THREAD

**Issue by iovdin** [libvips/libvips#1073]
https://storage.googleapis.com/frameright-crops/single_crop.png

``` 
$ vipsheader -a single_crop.png

single_crop.png: 512x512 uchar, 3 bands, srgb, pngload
width: 512
height: 512
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.834
yres: 2.834
filename: test/single_crop.png
vips-loader: pngload
```

``` 
$ exiftool single_crop.png

ExifTool Version Number         : 11.09
File Name                       : single_crop.png
Directory                       : test
File Size                       : 464 kB
File Modification Date/Time     : 2018:08:22 16:55:29+03:00
File Access Date/Time           : 2018:08:23 10:58:09+03:00
File Inode Change Date/Time     : 2018:08:22 17:37:31+03:00
File Permissions                : rw-r--r--
File Type                       : PNG
File Type Extension             : png
MIME Type                       : image/png
Image Width                     : 512
Image Height                    : 512
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
XMP Toolkit                     : Adobe XMP Core 5.6-c138 79.159824, 2016/09/14-01:09:01
Modify Date                     : 2018:08:22 13:55:29.219Z
Create Date                     : 2018:08:22 13:55:29.219Z
Metadata Date                   : 2018:05:25 15:37:10+02:00
Data                            : eyJvcmlnaW5hbFNpemUiOnsid2lkdGgiOjUxMiwiaGVpZ2h0Ijo1MTJ9LCJjcm9wcyI6W3sieCI6NjQsInkiOjc0LCJ3aWR0aCI6MzY
3LCJoZWlnaHQiOjM1MiwibGlicmFyeUlkIjpudWxsLCJsaWJyYXJ5SXRlbUlkIjpudWxsLCJyYXRpbyI6eyJ4IjozNjcsInkiOjM1Mn0sImZpeGVkUmF0aW8iOnRydWUsIm1pblNp
emUiOnsid2lkdGgiOjM2NywiaGVpZ2h0IjozNTJ9LCJyZXFzIjpudWxsfV0sIm1ldGFEYXRhRm9ybWF0IjoxfQ==
SRGB Rendering                  : Perceptual
Image Size                      : 512x512
Megapixels                      : 0.262

```

**Comment by jcupitt** [comment]
I made a branch with a quick hack:

https://github.com/jcupitt/libvips/tree/add-xmp-png-support

it does XMP read from png, but not write. I added non-xmp text metadata as comment fields, perhaps that's OK.

**Comment by iovdin** [comment]
@jcupitt Thank you. 
I use libvips from lovell/sharp. I'm not sure I can handle custom branch building.
There is a workaround for my issue, so I can wait till it merges to master

**Comment by jcupitt** [comment]
It seems to all work now. It looks like exif is now supported in png with `png_get_eXIf_1()`. That should probably be a separate issue. 

This branch should get merged for libvips 8.8.

**Comment by jcupitt** [comment]
OK, there's a PR and it's all ready to be merged. I'll close this issue. Thanks for suggesting it!
